### DISCUSSION TITLE
Question about the formula used in computing Akinci12 fluid-solid coupling forces

### DISCUSSION THREAD

**Discussion by zhehaoli1999** [InteractiveComputerGraphics/SPlisHSPlasH#219] [Q&A]
Thank you so much for all the great work! 👍👍👍
I have a question about the stability of computing fluid-solid coupling forces:

In my experiment with DFSPH (with the warmup), I found that the Akinci12 fluid-solid coupling forces have big oscillations when a dynamic rigid body object falls into the fluid and then moves with the fluid, the curve of coupling forces plots like this:
![image](https://user-images.githubusercontent.com/47491676/207784261-dbf5be35-9f78-4e5c-9df6-130dea3eb19f.png)

I have tried other boundary handling methods in SPlisHSPlasH such as the volume map method but the oscillation still exists. I expect the coupling force to be smoother in reality, and **I think the oscillations may be related to the unstable pressure approximation at the boundary of fluid and solids** (If I am wrong please correct me, thx!) 

I notice in computing the Akinci12 fluid-solid coupling forces, you use the formula from the paper:
![image](https://user-images.githubusercontent.com/47491676/207781008-19947170-3478-4cbf-a4ae-3054deb3e5c9.png)

Instead of using the symmetric formula introduced in your SPH tutorial paper:
![image](https://user-images.githubusercontent.com/47491676/207781117-ad91ef15-6165-4478-9926-276e603cacf1.png)

So I'd like to know if it is better to use the symmetric formula to stabilize the pressure approximation at the boundary of fluid and solid? or even use more advanced techniques such as the MLS extrpolation of solid pressures in Band18 paper "MLS Pressure Extrapolation for the Boundary Handling in Divergence-Free SPH". 

Thanks for any reply! Really appreciate your help!

**Comment by janbender** [comment]
I think you see the oscillations since the simulation of the dynamic rigid bodies and the simulation of the fluid is only weakly coupled. To get better results you would need to either use smaller time steps or to implement a strong coupling like this one: 

https://animation.rwth-aachen.de/publication/0563/

Btw. an implementation of this method is already on our long ToDo list. 

Note that we do not use the symmetric formula since we also reformulated the method a bit. In our reformulation you don't need the second term anymore. I also don't think this will lead to less oscillations. Also the pressure extrapolation won't help as long as you have a weak coupling.

**Reply by zhehaoli1999** [reply]
Thanks for your reply!!

I have tried to use smaller time steps by reducing the timestep from 0.005 to 0.0001 (DFSPH with divergenceSolve and warm start), however the oscillation still exist like the plot below: 
![image](https://user-images.githubusercontent.com/47491676/208807444-5472689a-4adf-462c-88f2-950b055e1926.png)

Then I guess the root cause for the oscillation in small timesteps may still be the weak coupling scheme. 

I have already read the strong coupling paper which you recommend, and I think it seems not very difficult to implement if there is only one dynamic rigid body in the scene without colliding with static walls, 
in which case I think all I need to do is to update the velocity of the rigid body **during**, in stead of **after**, the pressure solve and divergence solve in DFSPH  (If I am wrong please correct me, thx!).  I will try this and report the result soon.
